packages/taiko-client/pkg/rpc/utils.go

```diff
@@ -248,6 +248,126 @@ func GetBlockProofStatus(
 	}, nil
 }
 
+// BatchGetBlocksProofStatus checks whether the batch of L2 blocks still need new proofs or new contests.
+// Here are the possible status:
+// 1. No proof on chain at all.
+// 2. A valid proof has been submitted.
+// 3. An invalid proof has been submitted, and there is no valid contest.
+// 4. An invalid proof has been submitted, and there is a valid contest.
+func BatchGetBlocksProofStatus(
+	ctx context.Context,
+	cli *Client,
+	ids []*big.Int,
+	proverAddress common.Address,
+	proverSetAddress common.Address,
+) ([]*BlockProofStatus, error) {
+	ctxWithTimeout, cancel := CtxWithTimeoutOrDefault(ctx, defaultTimeout)
+	defer cancel()
+	var (
+		parentHashes = make([][32]byte, len(ids))
+		parents      = make([]*types.Header, len(ids))
+		blockIDs     = make([]uint64, len(ids))
+		result       = make([]*BlockProofStatus, len(ids))
+	)
+	// Get the local L2 parent header.
+	g, gCtx := errgroup.WithContext(ctxWithTimeout)
```

Comment:
And after thinking twice, let's do batch requests https://geth.ethereum.org/docs/interacting-with-geth/rpc/batch here, to make the network requests always `O1`.

<img width="822" alt="image" src="https://github.com/user-attachments/assets/c47b005d-67e2-42f5-9f41-ff6bb5dbebea">

Reply:
OK, pretty good!